### PR TITLE
Particles2D: set emitting if not set before generating Visibility Rect

### PR DESCRIPTION
**Pull request by natrim**
ie. made changes from #20977 for Particles2D too

edit: re-pushed to fix clang style warning

### CODE REVIEW

File: editor/plugins/particles_2d_editor_plugin.cpp [hunk 1]
```diff
@@ -90,6 +95,13 @@ void Particles2DEditorPlugin::_generate_visibility_rect() {
 
 	EditorProgress ep("gen_aabb", TTR("Generating AABB"), int(time));
 
+	bool was_emitting = particles->is_emitting();
+	if (!was_emitting) {
+		particles->set_emitting(true);
+		OS::get_singleton()->delay_usec(1000);
+	}
+	running = 0.0;
```
Comment (akien-mga) [inline editor/plugins/particles_2d_editor_plugin.cpp]: `running` is already 0, it's defined above and not modified. This was also a mistake in #20977, could you remove both?

Comment (natrim) [reply editor/plugins/particles_2d_editor_plugin.cpp]: @akien-mga removed

**Comment by akien-mga** [comment]
Thanks, could you now squash the commits? :)

**Comment by natrim** [comment]
@akien-mga squished .)

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**